superset-frontend/src/explore/components/AdhocFilterOption.jsx

```diff
@@ -60,7 +60,6 @@ class AdhocFilterOption extends React.PureComponent {
     // isNew is used to indicate whether to automatically open the overlay
     // once the overlay has been opened, the metric/filter will never be
     // considered new again.
-    this.props.adhocFilter.isNew = false;
```

Comment:
mutating a prop? that wasn't ok!

Reply:
As I wrote in https://github.com/apache/incubator-superset/pull/11428, although mutating a prop certainly isn't ok, it might not be that simple to fix it. I encountered many problems with auto-opening popup when I tried to copy your solution or to make a new state variable. I think replacing that ugly line requires a more serious refactor

Reply:
Mutating the prop of the object passed to the component was the simplest and most efficient way to update the select option state without forcing rerenders, and it's probably not as bad as it seems because we know no other places depend on this prop.

If you want to eliminate the need of mutating a prop, you should propagate the updated `adhocFilter` back to the select options by triggering the `onChange` event on `AdhocFilterControl` (which could have its own problem because of rerenders).

Reply:
I tried triggering `onChange` and just like you said, rerenders caused a whole bunch of new issues. So I think we can keep it as it is

Reply:
Ok, punted on changes to that file in favor of https://github.com/apache/incubator-superset/pull/11428